test/e2e/providers/vsphere/vsphere.go

```diff
@@ -133,9 +139,50 @@ func (p *Provider) GetType() config.PlatformType {
 }
 
 func (p *Provider) StorageSupport() bool {
-	return false
+	return true
 }
 
-func (p *Provider) CreatePVC(_ client.Interface, _ string) (*core.PersistentVolumeClaim, error) {
-	return nil, fmt.Errorf("storage not supported on vSphere")
+// CreatePVC creates a PVC for a dynamically provisioned volume
+func (p *Provider) CreatePVC(client client.Interface, namespace string) (*core.PersistentVolumeClaim, error) {
+	// Use a StorageClass to allow for dynamic volume provisioning
+	// https://docs.openshift.com/container-platform/4.12/storage/dynamic-provisioning.html#about_dynamic-provisioning
+	sc, err := p.ensureStorageClass(client)
+	if err != nil {
+		return nil, fmt.Errorf("unable to ensure a usable StorageClass is created: %w", err)
+	}
+	pvcSpec := core.PersistentVolumeClaim{
+		ObjectMeta: meta.ObjectMeta{
+			GenerateName: "e2e-",
+		},
+		Spec: core.PersistentVolumeClaimSpec{
+			AccessModes: []core.PersistentVolumeAccessMode{core.ReadWriteOnce},
```

Comment:
Shouldn't we be testing `ReadWriteMany` as we have had requests for that?

Reply:
ReadWriteMany requires the vSphere env to support it. VMC does not support it.
Potentially we could try to determine if the workload is running in an environment that does support it, and use it in that case. I'm not sure how to do that, and would prefer to not have to add that extra work into this PR which is a release blocker.

Reply:
I am fine not doing that in this PR but we need to track getting this to work in VMC. Maybe a SPLAT ticket? We don't want to wait for a customer issue to be opened to enable this in VMC.